proptest/src/lib.rs

```diff
@@ -96,6 +96,7 @@ pub mod tuple;
 pub mod option;
 pub mod result;
 pub mod sample;
+pub mod state_machine;
```

Comment:
After more thought, I think this is a good idea. I don't think there should be any issue with separating it, I'll give it a try

Reply:
There are some symbols that are currently `pub(crate)` that would need to be made public:

- `bits::varsize::VarBitSet`
- `collection::SizeRange::start_end_incl`
- `num::sample_uniform_incl` - this one is a pretty tiny indirection

wdyt?

Reply:
> bits::varsize::VarBitSet

this one seems fine to me

> collection::SizeRange::start_end_incl

this one also seems fine

> sample_uniform_incl

the impl of this ones only one line so i'd probably just call that directly rather than exposing it but i dont have strong opinions.
```
Uniform::new_inclusive(start, end).sample(run.rng())
```

Reply:
alright, there's proptest-state-machine now (done in https://github.com/proptest-rs/proptest/pull/310/files/4178671c9ea77b29f0110cac4ac5a88fc99d2121..a1dc279667001c0cce9bafa5f1b9e3d47bc9cc24)